packages/kcmsf/src/pages/matchResult.tsx

```diff
@@ -22,94 +22,193 @@ export const MatchResult = () => {
           setRightTeamResult(match.runResults[1]);
           setLeftTeamResult(match.runResults[0]);
         }
-      } else {
-        //Todo:本選の場合は合計を出す
-        //loopとかを使って合計を計算する?
-        // console.log(match.runResults);
-        // console.log(matchInfo?.teams.right?.id);
-        // const rightTeamResultPoints=match.runResults
-        //   .filter((result) => {
-        //     result.teamID == matchInfo?.teams.right?.id;
-        //   })
-        //   .reduce((sum, result) => sum + result.points,0);
-        // console.log(rightTeamResultPoints);
+      } else if (matchInfo?.matchType == "main") {
+        const team1ID = match.team1.id;
+        const team2ID = match.team2.id;
 
-        if (match.runResults[0].teamID == matchInfo?.teams.right?.id) {
-          setRightTeamResult(match.runResults[0]);
-          setLeftTeamResult(match.runResults[1]);
-        } else {
-          setRightTeamResult(match.runResults[1]);
-          setLeftTeamResult(match.runResults[0]);
-        }
+        const team1Results = match.runResults.filter(
+          (result) => result.teamID === team1ID
+        );
+        const team2Results = match.runResults.filter(
+          (result) => result.teamID === team2ID
+        );
+
+        const team1Point = team1Results.reduce(
+          (sum, result) => sum + result.points,
+          0
+        );
+        const team2Point = team2Results.reduce(
+          (sum, result) => sum + result.points,
+          0
+        );
+
+        const team1GoalTime = team1Results
+          .map((result) => result.goalTimeSeconds)
+          .sort()[1];
+        const team2GoalTime = team2Results
+          .map((result) => result.goalTimeSeconds)
+          .sort()[1];
+
+        const team1Result = {
+          teamID: team1ID,
+          points: team1Point,
+          goalTimeSeconds: team1GoalTime,
+        } as RunResult;
```

Comment:
アサーションより変数の型を明示した方が良さそう
```suggestion
        const team1Result: RunResult = {
          teamID: team1ID,
          points: team1Point,
          goalTimeSeconds: team1GoalTime,
        };
```